actors/builtin/miner/testing.go

```diff
@@ -201,14 +201,30 @@ func CheckDeadlineStateInvariants(deadline *Deadline, store adt.Store, quant Qua
 	})
 	acc.RequireNoError(err, "error iterating partitions")
 
-	// Check PoSt submissions
-	if postSubmissions, err := deadline.PostSubmissions.All(1 << 20); err != nil {
-		acc.Addf("error expanding post submissions: %v", err)
-	} else {
-		for _, p := range postSubmissions {
-			acc.Require(p <= partitionCount, "invalid PoSt submission for partition %d of %d", p, partitionCount)
+	// Check partitions snapshot to make sure we take the snapshot after
+	// dealing with recovering power and unproven power.
+	partitionsSnapshot, err := deadline.PartitionsSnapshotArray(store)
+	acc.RequireNoError(err, "error loading partitions snapshot")
+	err = partitionsSnapshot.ForEach(&partition, func(i int64) error {
+		acc := acc.WithPrefix("partition snapshot %d: ", i) // Shadow
+
+		acc.Require(partition.RecoveringPower.IsZero(), "snapshot partition has recovering power")
+		if noRecoveries, err := partition.Recoveries.IsEmpty(); err != nil {
+			acc.Addf("error counting recoveries: %v", err)
+		} else {
+			acc.Require(noRecoveries, "snapshot partition has pending recoveries")
 		}
-	}
+
+		acc.Require(partition.UnprovenPower.IsZero(), "snapshot partition has unproven power")
+		if noUnproven, err := partition.Unproven.IsEmpty(); err != nil {
+			acc.Addf("error counting unproven: %v", err)
+		} else {
+			acc.Require(noUnproven, "snapshot partition has unproven sectors")
+		}
+
+		return nil
+	})
+	acc.RequireNoError(err, "error iterating partitions snapshot")
```

Comment:
There must be some more invariants about the post submissions, and things that could be broken if compaction and termination blackout windows were not effective.

Reply:
So, unfortunately, I don't know the current epoch from this function. However, I could just clear the partitions/proofs snapshots on compaction. That's probably the most correct approach.

Reply:
The epoch can be plumbed here if helpful (it was used in a prior migration/validation).

But I see what you mean, it's hard to assert more without knowing about the window. I'm nervous about clearing the snapshots: code will probably assume they exist, and if they can fail to exist then everywhere should check for and handle that, even if we think that epoch windows should prevent ever discovering them missing.

Ok to leave this for now.